command/agent/event_endpoint.go

```diff
@@ -57,7 +61,13 @@ func (s *HTTPServer) EventFire(resp http.ResponseWriter, req *http.Request) (int
 	}
 
 	// Try to fire the event
-	if err := s.agent.UserEvent(dc, event); err != nil {
+	if err := s.agent.UserEvent(dc, token, event); err != nil {
+		if strings.Contains(err.Error(), permissionDenied) {
```

Comment:
I think at some point we need to do a pass-through and have some kind of error struct which provides its type. I agree this is sort of ghetto, we should definitely address that broadly in a separate PR.